Geometry_Engine/Query/IsContaining.cs

```diff
@@ -709,6 +709,182 @@ public static bool IsContaining(this Cuboid cuboid, IGeometry geometry, bool acc
+
+            List<Point> ctrPoints = panels.SelectMany(x => x.IControlPoints()).ToList();
+            BoundingBox boundingBox = BH.Engine.Geometry.Query.Bounds(ctrPoints);
+
+            List<bool> areContained = new List<bool>();
+
+            foreach (List<Point> pts in pointLists)
+            {
+                bool isContained = false;
+                if (acceptPartialContainment)
+                    isContained = pts.Any(point => IsContaining(panels, planes, boundingBox, point, acceptOnEdges));
+                else
+                    isContained = pts.All(point => IsContaining(panels, planes, boundingBox, point, acceptOnEdges));
+                areContained.Add(isContained);
+            }
+
+            return areContained;
+        }
+
+        /***************************************************/
+        /**** Private Methods - Closed Volume Polylines ****/
+        /***************************************************/
+
+        [Description("Checks if a point lies within a collection of polylines describing a closed volume using their primitive planes and bounds.")]
+        [Input("panels", "A collection of Polylines outlining each of the faces of the closed volume to check with.")]
+        [Input("planes", "Planes corresponding to each panel for intersection calculations.")]
+        [Input("boundingBox", "The bounding box of the collection of panels.")]
+        [Input("point", "The point to check to see if it is contained within the bounds of the panels.")]
+        [Input("acceptOnEdges", "Decide whether to allow the point to sit on the edge of the panel, default false.")]
+        [Input("tolerance", "Distance tolerance to use to determine intersections.")]
+        [Output("isContaining", "True if the point is contained within the bounds of the panels, false if it is not for each point provided.")]
+        private static bool IsContaining(this List<Polyline> panels, List<Plane> planes, BoundingBox boundingBox, Point point, bool acceptOnEdges = false, double tolerance = BH.oM.Geometry.Tolerance.Distance)
+        {
+            //Return if point is null even without checking boundingBox.IsContaining(point)
+            if (point == null)
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a collection of panels contains a point if the point is null.");
+                return false;
+            }
+
+            if (!BH.Engine.Geometry.Query.IsContaining(boundingBox, point, true, tolerance))
+                return false;
+
+            //We need to check one line that starts in the point and end outside the bounding box
+            Vector vector = new Vector() { X = 1, Y = 0, Z = 0 };
+            //Use a length longer than the longest side in the bounding box. Change to infinite line?
+            Line line = new Line() { Start = point, End = point.Translate(vector * (((boundingBox.Max - boundingBox.Min).Length()) * 10)) };
+
+            bool isInPlane = false;
+            do
```

Comment:
What if there point lies in planes along `(1,0,0)` and `(0.5,0.5,0.5)` at the same time? Won't this loop go forever?

Actually, can't you simply take an arbitrary vector that has absolute value of dot product with each of plane normal larger than tolerance?

Reply:
fixed